### PR TITLE
OMEMO speedup / fixes

### PR DESCRIPTION
**Pull request by FriedrichAltheide**

### CODE REVIEW

File: Monal/Classes/ContactDetails.m [hunk 1]
```diff
@@ -523,9 +523,8 @@ -(void) refreshMute
 -(IBAction) toggleEncryption:(id)sender
 {
 #ifndef DISABLE_OMEMO
-    NSArray* devices = [self.xmppAccount.omemo knownDevicesForAddressName:self.contact.contactJid];
     [MLChatViewHelper<ContactDetails*>
-        toggleEncryptionForContact:self.contact withKnownDevices:devices withSelf:self afterToggle:^() {
+        toggleEncryptionForContact:self.contact withSelf:self afterToggle:^() {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/ContactDetails.m:527]: not needed anymore, ContactDetails.m will be removed soon (for next beta I think), doesn't hurt though

File: Monal/Classes/MLChatViewHelper.m [hunk 2]
```diff
@@ -12,8 +12,12 @@
 
 @implementation MLChatViewHelper
 
-+(void) toggleEncryptionForContact:(MLContact*) contact withKnownDevices:(NSArray*) knownDevices withSelf:(id) andSelf afterToggle:(void (^)(void)) afterToggle {
-    if(knownDevices.count == 0 && contact.isEncrypted == NO) {
++(void) toggleEncryptionForContact:(MLContact*) contact withSelf:(id) andSelf afterToggle:(void (^)(void)) afterToggle {
+    // Update the encryption value in the caller class
+    BOOL prevEncryptionState = contact.isEncrypted;
+    contact.isEncrypted = [contact toggleEncryption:!contact.isEncrypted];
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLChatViewHelper.m]: this will be written by toggleEncryption already, no need to update it twice

Comment (tmolitor-stud-tu) [reply Monal/Classes/MLChatViewHelper.m]: the return value indicates, if the toggle was successful, not if encryption was enabled!

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]
I don't really understand the stuff in MLOMEMO.m, but the rest is fine (minus my comments)

File: Monal/Classes/MLChatViewHelper.m [hunk 3]
```diff
@@ -12,8 +12,9 @@
 
 @implementation MLChatViewHelper
 
-+(void) toggleEncryptionForContact:(MLContact*) contact withKnownDevices:(NSArray*) knownDevices withSelf:(id) andSelf afterToggle:(void (^)(void)) afterToggle {
-    if(knownDevices.count == 0 && contact.isEncrypted == NO) {
++(void) toggleEncryptionForContact:(MLContact*) contact withSelf:(id) andSelf afterToggle:(void (^)(void)) afterToggle {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLChatViewHelper.m]: style: `{` in next line

File: Monal/Classes/MLChatViewHelper.m [hunk 4]
```diff
@@ -12,8 +12,9 @@
...
+    // Update the encryption value in the caller class
+    if(![contact toggleEncryption:!contact.isEncrypted]) {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLChatViewHelper.m]: style again

File: Monal/Classes/MLContact.m [hunk 5]
```diff
@@ -376,8 +376,11 @@ -(BOOL) toggleEncryption:(BOOL) encrypt
     if(account == nil || account.omemo == nil)
         return NO;
     NSArray* knownDevices = [account.omemo knownDevicesForAddressName:self.contactJid];
-    if(encrypt && !self.isEncrypted && knownDevices.count == 0)
+    if(!self.isEncrypted && encrypt && knownDevices.count == 0) {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLContact.m]: style

Comment (tmolitor-stud-tu) [reply Monal/Classes/MLContact.m]: why did you reorder this? :D

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]
approved once the style is correct

File: Monal/Classes/MLContact.m (hunk 5 above)
Comment (FriedrichAltheide) [reply Monal/Classes/MLContact.m]: old, new, condition

**Review by tmolitor-stud-tu — APPROVED** [review]
perfect

**Pull request merged**